### ISSUE TITLE
Remove redundant LoadImagesandLabels output

### ISSUE THREAD

**Issue by NanoCode012** [ultralytics/yolov5#2184]
## ❔Question
When training, I keep noticing the duplicates when loading images and labels. I think it boils down to this `tqdm` but was wondering whether someone could figure it out. 

https://github.com/ultralytics/yolov5/blob/404749a33cc29d119f54b2ce35bf3b33a847a487/utils/datasets.py#L385-L386

## Additional context
Env: Docker + Jupyterlab
Command: `python train.py --device cpu --epoch 3`

![image](https://user-images.githubusercontent.com/9899957/107671281-3eed5e00-6cc6-11eb-9c5c-b1fdf893e11a.png)

**Comment by glenn-jocher** [comment]
@NanoCode012 yes this is a DDP Multi-GPU printout I think. Basically each of the nodes needs to load the training data (but not the val data?), so the training set at least will appear 8 times for 8 nodes. val should only appear once though I think, not sure why yours shows up there 7 times it looks like.

**Comment by NanoCode012** [comment]
Hi @glenn-jocher , this is a CPU run on docker not multi-gpu run. `python train.py --epoch 3 --device cpu`. 

I get similar output on single GPU run. Do you get the same problem?

Edit: I got the below on colab. Is it to be expected?

![image](https://user-images.githubusercontent.com/9899957/107690141-4c154780-6cdc-11eb-9163-e075307ed1c8.png)

**Comment by glenn-jocher** [comment]
@NanoCode012 yes, this is normal the first time a dataset is scanned. The second time there should be fewer lines. What happens is the scan tqdm appears in realtime, and then the cache is saved, and then the cache is read, printing an instant tqdm line to simulate the earlier scanning results. The first scan will appear redundant in this case. Maybe there's a better way to handle this, the code is here:
https://github.com/ultralytics/yolov5/blob/404749a33cc29d119f54b2ce35bf3b33a847a487/utils/datasets.py#L373-L388

The steps are here, the redundancies are when both steps run.
1. If no cache, then scan all images and labels, and create new *.cache (prints REALTIME Scanning... tqdm)
2. Load saved *.cache (prints SAVED Scanning... tqdm)

**Comment by glenn-jocher** [comment]
@NanoCode012 there's probably a better way to handle this, so the L386 tqdm message only prints if a good *.cache file already exists, otherwise it will be the second printing of the same information.

**Comment by glenn-jocher** [comment]
@NanoCode012 I've integrated a fix for this in commit https://github.com/ultralytics/yolov5/pull/2188/commits/d11487ecdca38309924d8ee638d3c7ada4472d1b of the current segmentation-loader PR. This should prevent duplicates printings:

<img width="1000" alt="Screen Shot 2021-02-11 at 7 55 13 PM" src="https://user-images.githubusercontent.com/26833433/107728504-1dc74600-6ca3-11eb-8cf4-dab888235691.png">

EDIT: I see have the booleans backwards, will fix.

**Comment by glenn-jocher** [comment]
@NanoCode012 ok, PR is merged and I think the redundant outputs have now disappeared. I've updated the Colab notebook also to show the cleaned up output (you can see this in the Test and Train cells):
https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb
